### PR TITLE
feat(insured-bridge-relayer): add speed up bridging and additional standardization

### PR DESCRIPTION
**Pull request by chrismaree**
**Summary**

This PR implements two main things:
1) adds the `speedUpRelay` implementation and tests
2) adds a number of renames in some state variables in the bridge contracts. primarily:
2.a) `recipient` is renamed to `l1Recipient` to be more consistent with the other state variables
2.b) `timestamp` in the OVM contracts is renamed to `depositTimestamp` to match with how this variable is used in other places

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/financial-templates-lib/test/clients/InsuredBridgeL2Client.js [hunk 1]
```diff
@@ -143,13 +143,13 @@ describe("InsuredBridgeL2Client", () => {
...
       depositHash: "",
-      timestamp: depositTimestamp2,
-      sender: user1,
-      recipient: user2,
+      depositTimestamp: depositTimestamp2,
+      l1Recipient: user2,
+      l2Sender: user1,
       l1Token: l1TokenAddress,
       amount: depositAmount,
-      slowRelayFeePct: slowRelayFeePct,
```
Comment (chrismaree) [inline packages/financial-templates-lib/test/clients/InsuredBridgeL2Client.js:151]: syntax clean up

File: packages/insured-bridge-relayer/src/Relayer.ts [hunk 2]
```diff
@@ -196,12 +196,36 @@ export class Relayer {
...
       else throw receipt;
     } catch (error) {
-      this.logger.error({ at: "InsuredBridgeRelayer#Relayer", type: "Something went wrong slow relaying!", error });
+      this.logger.error({ at: "InsuredBridgeRelayer#Relayer", type: "Something errored slow relaying!", error });
     }
   }
 
-  private async speedUpRelay(/* deposit: Deposit*/) {
-    // TODO: implement
+  private async speedUpRelay(deposit: Deposit) {
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/Relayer.ts:203]: main implementation

File: packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol [hunk 3]
```diff
@@ -61,8 +61,8 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
         uint64 depositId;
         uint64 depositTimestamp;
         address l2Sender;
-        address recipient;
-        address l1Token;
+        address l1Recipient;
+        address l1Token; // todo: we can remove this prop. it's not needed as this contract has a unique l1 token.
```
Comment (nicholaspai) [inline packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol:65]: +1

File: packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol [hunk 4]
```diff
@@ -80,9 +80,9 @@ contract OVM_BridgeDepositBox is OVM_CrossDomainEnabled, OVM_Testable {
     // TODO: change the order of these to match the way they are used in the bridge pool.
     event FundsDeposited(
         uint256 depositId,
-        uint256 timestamp,
-        address sender,
-        address recipient,
+        uint256 depositTimestamp,
```
Comment (nicholaspai) [inline packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol:83]: FYI I think we're going to remove `depositTimestamp` in #3363

File: packages/insured-bridge-relayer/src/Relayer.ts [hunk 5]
```diff
@@ -196,12 +196,36 @@ export class Relayer {
...
+        web3: this.l1Client.l1Web3,
+        transaction: (SpeedUpRelayTx as unknown) as TransactionType,
+        transactionConfig: { gasPrice: this.gasEstimator.getCurrentFastPrice().toString(), from: this.account },
+        availableAccounts: 1,
+      });
+
+      if (receipt.events)
+        this.logger.info({
+          at: "InsuredBridgeRelayer#Relayer",
+          type: "Relay instantly sped up 🚀",
```
Comment (nicholaspai) [inline packages/insured-bridge-relayer/src/Relayer.ts:219]: nit: isn't it redundant to say the relay was instantly sped up? Why not just say "Relay sped up"

File: packages/insured-bridge-relayer/src/Relayer.ts [hunk 6]
```diff
@@ -196,12 +196,36 @@ export class Relayer {
...
+          tx: receipt.transactionHash,
+          depositHash: receipt.events.RelaySpedUp.returnValues.depositHash,
+          instantRelayer: receipt.events.RelaySpedUp.returnValues.instantRelayer,
+          transactionConfig,
+        });
+      else throw receipt;
```
Comment (nicholaspai) [inline packages/insured-bridge-relayer/src/Relayer.ts:225]: Never seen this before where you just `throw object`. Why not throw a customized error message

**Review by nicholaspai — APPROVED** [review]
image

LGTM good variable renaming and i agree with the speed up relay impl.

**Pull request merged**